execution/executor/src/components/chunk_output.rs

```diff
@@ -31,16 +33,78 @@ use aptos_vm::{
 };
 use fail::fail_point;
 use move_core_types::vm_status::StatusCode;
-use once_cell::sync::Lazy;
-use std::{ops::Deref, sync::Arc, time::Duration};
+use once_cell::sync::{Lazy, OnceCell};
+use std::{
+    net::{IpAddr, Ipv4Addr, SocketAddr},
+    ops::Deref,
+    sync::Arc,
+    time::Duration,
+};
+
+static REMOTE_SHARDING: OnceCell<bool> = OnceCell::new();
+static REMOTE_ADDRESSES: OnceCell<Vec<SocketAddr>> = OnceCell::new();
+static COORDINATOR_ADDRESS: OnceCell<SocketAddr> = OnceCell::new();
+
+/// Sets the flag when invoked the first time.
+pub fn set_remote_sharding(enable: bool) {
```

Comment:
This flag is not needed, just set remote sharding to enabled when `remote_addresses` is set.